Sources/Sentry/SentryProfiler.mm

```diff
@@ -354,9 +354,12 @@ - (void)appendBacktrace:(const Backtrace &)backtrace
         }
         if (queueAddress != nil && state.queueMetadata[queueAddress] == nil
             && backtrace.queueMetadata.label != nullptr) {
-            state.queueMetadata[queueAddress] = @ {
-                @"label" : [NSString stringWithUTF8String:backtrace.queueMetadata.label->c_str()]
-            };
+            NSString *const labelNSStr =
+                [NSString stringWithUTF8String:backtrace.queueMetadata.label->c_str()];
+            // -[NSString stringWithUTF8String:] can return `nil` for malformed string data
```

Comment:
Here's a repro of this happening:
<img width="1232" alt="CleanShot 2023-05-23 at 22 48 47@2x" src="https://github.com/getsentry/sentry-cocoa/assets/353158/d7216813-dbec-4274-9042-b7e4d3fa6ad6">